extensions/commands/recipe/cmd_create_sbom.py

```diff
@@ -0,0 +1,141 @@
+        type=package_type_to_component_type(n.conanfile.package_type),
+        name=name_,
+        version=n.conanfile.version,
+        licenses=licenses(n.conanfile.license),
+        bom_ref=purl.to_string(),
+        purl=purl,
+        description=n.conanfile.description
+    )
+    if n.conanfile.homepage:
+        result.external_references.add(ExternalReference(
+            type=ExternalReferenceType.WEBSITE,
+            url=XsUri(n.conanfile.homepage),
+        ))
+    return result
+
+
+def format_cyclonedx_14_json(result):
+    from cyclonedx.output.json import JsonV1Dot4
+    serialized_json = JsonV1Dot4(result).output_as_string()
+    cli_out_write(serialized_json)
+
+
+def format_text(result):
+    out = ConanOutput()
+    out.error("Format 'text' not supported")
+
+
+@conan_command(group="Recipe", formatters={
+    "text": format_text,  # added by default in BaseConanCommand.__init__
+    "cyclonedx_1.4_json": format_cyclonedx_14_json})
+def create_sbom(conan_api: ConanAPI, parser, *args):
+    """
+    creates an SBOM in CycloneDX 1.4 JSON format
+    """
+    try:
+        import cyclonedx
+    except ModuleNotFoundError:
+        sys.stderr.write(
+            "The sbom extension needs an additional package, please run 'pip install cyclonedx-python-lib'\n")
+        sys.exit(1)
+    from cyclonedx.model.bom import Bom
+    # BEGIN COPY FROM conan: cli/commands/graph.py
+    common_graph_args(parser)
+    args = parser.parse_args(*args)
+    validate_common_graph_args(args)
+    cwd = os.getcwd()
+    path = conan_api.local.get_conanfile_path(args.path, cwd, py=None) if args.path else None
+    remotes = conan_api.remotes.list(args.remote) if not args.no_remote else []
+    overrides = eval(args.lockfile_overrides) if args.lockfile_overrides else None
+    lockfile = conan_api.lockfile.get_lockfile(lockfile=args.lockfile,
+                                               conanfile_path=path,
+                                               cwd=cwd,
+                                               partial=args.lockfile_partial,
+                                               overrides=overrides)
+    profile_host, profile_build = conan_api.profiles.get_profiles_from_args(args)
+    if path:
+        deps_graph = conan_api.graph.load_graph_consumer(path, args.name, args.version,
+                                                         args.user, args.channel,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update)
+    else:
+        deps_graph = conan_api.graph.load_graph_requires(args.requires, args.tool_requires,
+                                                         profile_host, profile_build, lockfile,
+                                                         remotes, args.update)
+    # END COPY
+    name.unknown_name_created = False
+    components = {n: create_component(n) for n in deps_graph.nodes}
```

Comment:
maybe use another way to define the dict's keys. an object for a key might lead to runtime-dependent issues.

Reply:
What do you have in mind? The only thing I can think of is `id(n)`, which is a bit ugly

Reply:
> A dictionary’s keys are almost arbitrary values. Values that are not hashable, that is, values containing lists, dictionaries or other mutable types (that are compared by value rather than by object identity) may not be used as keys.

So, as long as the has does not change, you are fine, i guess.
I am no expert on this specific subject here.

did some tests, objects for keys seams fine in py3.7
test script
```py
o1 = object()
o2 = object()
d = {o1:1, o2:2}
for e in d.keys():
  print(type(e), repr(e))
```